Sources/OpenAI/Public/Utilities/CodableUtilities.swift

```diff
@@ -0,0 +1,118 @@
+//
+//  CodableUtilities.swift
+//
+//
+//  Created by Federico Vitale on 09/11/23.
+//
+
+import Foundation
+
+
+/// Allows having dynamic keys in codables. 
+struct DynamicKey: CodingKey {
+    var stringValue: String
+    var intValue: Int?
+    
+    init(stringValue: String) {
+        self.stringValue = stringValue
+    }
+    
+    init?(intValue: Int) {
+        self.intValue = intValue
+        self.stringValue = "\(intValue)"
+    }
+}
+
+
+///  Allows to encode/decode ``Chat`` or ``Codable`` (T)
+///     ```swift
+///     struct Person: Codable, Equatable {
+///         let name: StringOrCodable<FullName>
+///
+///         struct FullName {
+///             firstName: String
+///             lastName: String
+///         }
+///     }
+///
+///     let person = Person(name: .string("John Doe"))
+///     let fullNamePerson = Person(name: .object(.init(firstName: "John", lastName: "Doe")))
+///     ```
+public enum StringOrCodable<T: Codable>: Equatable, Codable where T: Equatable {
```

Comment:
[Looking at the api](https://platform.openai.com/docs/api-reference/chat/create) looks like only the role `user` can actually send arrays.


![CleanShot 2023-11-14 at 12 13 57@2x](https://github.com/MacPaw/OpenAI/assets/16429579/1a09c09e-2af6-4f2e-8447-eaa64ad510c9)
> ChatContent with `user` role

![CleanShot 2023-11-14 at 12 13 33@2x](https://github.com/MacPaw/OpenAI/assets/16429579/8108435c-b705-4a10-8a54-0400cf4838d2)
> ChatContent with `assistant` role

This should be fine with the actual implementation, I tried to handle both via `StringOrCodable`

Reply:
i'm not sure if there's a cleaner way to handle this